lib/Sema/CSGen.cpp

```diff
@@ -2874,7 +2881,31 @@ namespace {
     }
 
     Type visitPackExpansionExpr(PackExpansionExpr *expr) {
-      llvm_unreachable("not implemented for PackExpansionExpr");
+      for (auto *binding : expr->getBindings()) {
+        auto type = visit(binding);
```

Comment:
This might be an oversight - `Traversal::visitPackExpansionExpr` in ASTWalker.cpp needs to be updated to walk into `bindings` and re-set them, same applies to ASTDumper and ASTPrinter most likely.

Reply:
No, not visiting these in ASTWalker was intentional. This follows the same pattern as `OpenExistentialExpr` which only visits the subexpression containing opaque values, and the bindings are handled separately where needed. These aren't actually separate values in the AST, but we do need the types in the constraint system.

Reply:
Ok, we probably need a comment about this then... ASTDumper still needs to be updated to print bindings and opaque values just like `OpenExistentialExpr` does.